### ISSUE TITLE
how to monitor stage 1 training

### ISSUE THREAD

**Issue by yuvaramsingh94** [wangz10/contrastive_loss#2]
Hi ,
   Thanks for this implimentation . i like to know how to monitor stage 1 training . i am applying this concept for a custom data for image classification. i the visualized projection are not getting grouped together even after running for 50-60 epoch . can you share some idea on how to monitor. also my batch size id limited to 12 because of GPU limitation . do you think this is am issue ?

thanks

**Comment by wangz10** [comment]
@yuvaramsingh94  I think you can monitor the learning curves for stage 1 training to see if the loss gradually decline over epoches. 

It would require some extra effort to quantitatively evaluate the embeddings. Many clustering metrics, such as AMI, ARI should work. Also considering using nearest neighbor generalization error, which was used to evaluate t-SNE. Checkout details here: https://github.com/MaayanLab/Graph-DR/blob/master/DR-2019.ipynb

For NT-Xent loss, large batch size will help. Other loss functions may not benefit from large batch sizes.
